### ISSUE TITLE
Update documentation to always refer to Project ID for "Project" arguments

### ISSUE THREAD

**Issue by nwatnick** [terraform-providers/terraform-provider-google#3419]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Documentation does not consistently suggest using the Project ID for Project arguments on multiple resources & data sources.

For example:
Good: https://www.terraform.io/docs/providers/google/r/storage_bucket.html
"project - (Optional) The ID of the project in which the resource belongs. If it is not provided, the provider project is used."
Bad: https://www.terraform.io/docs/providers/google/d/google_container_cluster.html
"project - (Optional) The project in which the resource belongs. If it is not provided, the provider project is used."

Request: Can the documentation please be updated to use ID for Project consistently across all pages?

### References

This was done in the past for some resources with this PR:
https://github.com/terraform-providers/terraform-provider-google/pull/672
Looks like we need to review all resources to ensure this is updated everywhere in the docs.

**Comment by emilymye** [comment]
**tl;dr**: If the API accepts either, we won't explicitly document one

In most cases you can use either project ID or number, and this is pretty consistent across GCP docs as well when using `project`. Only a few resources _require_ id or number, and since it depends on the underlying GCP resource, this seems like something we probably won't put too much effort into fixing unless given specific resources that require only one format.
